plasmapy/formulary/dimensionless.py

```diff
@@ -483,3 +487,148 @@ def Mag_Reynolds(U: u.m / u.s, L: u.m, sigma: u.S / u.m) -> u.dimensionless_unsc
+
+    : `~astropy.units.UnitsWarning`
+        If units are not provided, SI units are assumed.
+
+    Raises
+    ------
+    `~plasmapy.utils.exceptions.RelativityError`
+        If the Alfvén velocity is greater than or equal to the speed of light.
+
+    `TypeError`
+        If ``B`` and/or ``density`` are not of type `~astropy.units.Quantity`,
+        or convertible.
+
+    `TypeError`
+        If ``ion`` is not of type or convertible to `~plasmapy.particles.particle_class.Particle`.
+
+    `TypeError`
+        If ``z_mean`` is not of type `int` or `float`.
+
+    `~astropy.units.UnitTypeError`
+        If the magnetic field ``B`` does not have units equivalent to
+        tesla.
+
+    `~astropy.units.UnitTypeError`
+        If the ``density`` does not have units equivalent to a number density
+        or mass density.
+
+    `ValueError`
+        If ``density`` is negative.
+
+    Notes
+    -----
+    For calculating the Alfvén speed `~plasmapy.formulary.speeds.Alfven_speed`
+    is used and for calculating the Lundquist number
+    `~plasmapy.formulary.dimensionless.magnetic_Reynolds` is used.
+
+    The Lundquist number is an important quantity in the study of
+    magnetic reconnection. For example, reconnection rates in both the
+    Sweet-Parker and Petschek models of magnetic reconnection can be expressed
+    in terms of the Lundquist number. In the Sweet-Parker model, a current
+    sheet with half-width :math:`L`, conductivity :math:`\sigma`, magnetic
+    diffusivity :math:`\eta = 1 / (\mu_0 \sigma)`, and Alfvén speed :math:`V_A`
+    at the inflow has a Lundquist number of
+    :math:`S = LV_A / \eta`. The dimensionless reconnection rate :math:`R`,
+    i.e., the ratio of the inflow to outflow speeds, can then be expressed as
+    :math:`R \sim 1 / \sqrt{S}`. Similarly, the maximum reconnection rate
+    in the Petschek model can be expressed as approximately
+    :math:`\pi / (8 \ln S)` :cite:p:`Priest:2000`.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> from astropy.constants.si import m_p, m_e
+    >>> L = 10**8 * u.m
+    >>> B = 10**2 * u.G
+    >>> n = 10**19 * u.m**-3
+    >>> rho = n*(m_p + m_e)
+    >>> sigma = 10**-7 * u.S / u.m
+    >>> Lundquist_number(L, B, rho, sigma)
+    <Quantity 0.86653839>
+    >>> Lundquist_number(L, B, n, sigma, ion="p")
+    <Quantity 0.86653839>
+    >>> Lundquist_number(L, B, n, sigma, ion="He +2")
+    <Quantity 0.43481967>
+    >>> Lundquist_number(L, B, n, sigma, ion="He", z_mean=1.8)
+    <Quantity 0.43476604>
+    >>> sigma = 10**-2 * u.S / u.m
+    >>> Lundquist_number(L, B, n, sigma, ion="He", z_mean=1.8)
+    <Quantity 43476.60420832>
+
+    Returns
+    -------
+    S : `~astropy.units.Quantity`
+        The Lundquist number.
+
+    """
+
+    alfven = speeds.Alfven_speed(B, density, ion=ion, z_mean=z_mean)
+    return Mag_Reynolds(alfven, L, sigma)
+
+
+Lu_ = Lundquist_number
+"""Alias to `~plasmapy.formulary.dimensionless.Lundquist_number`."""
```

Comment:
In the literature, _S_ is often used for the Lundquist number. However, I thought `S_` would be confusing as an alias since it is more commonly used for other things  (surfaces, entropy, Poynting flux, etc.). Instead I added `Lu_` which was used by my source in the docs, Priest & Forbes (2000), but not much elsewhere. So if it is alright to not have an alias here then I do not think `Lu_` should stay.